component/booksCard.jsx

```diff
@@ -0,0 +1,54 @@
+import useStore from '../hooks/useStore';
+import { useEffect } from 'react';
+import styled from 'styled-components';
+
+const Api = process.env.NEXT_PUBLIC_REACT_APP_API_KEY;
+
+export default function BooksCard({ apikey }) {
+  const fetchApi = useStore(state => state.fetchApi);
+  const fetchedData = useStore(state => state.fetchedData);
+
+  useEffect(() => {
+    fetchApi(
+      `https://api.nytimes.com/svc/books/v3/lists/current/hardcover-fiction.json?api-key=${apikey}`
+    );
+  }, [fetchApi]);
+  console.log(Api);
+  return (
+    <CardWrapper>
+      {fetchedData.results.books !== undefined ? (
```

Comment:
```suggestion
      {fetchedData?.results?.books !== undefined ? (
```
With this little adjustment your code does not crash if it cant find the results in your fetchedData. See `optional chaining opterator`.